### PR TITLE
Added QBFT information.

### LINKED ISSUE #710: Document QBFT
See Epic: https://app.zenhub.com/workspaces/team-groot-5e45ea08b6ec29767d67c876/issues/hyperledger/besu/1779

### PR DESCRIPTION
**Pull request by bgravenorst**
## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

## Describe the change

Added the QBFT  consensus engine

## Issue fixed

Fixes #710 

### For content changes

- [x] Doc content
- [x] Doc pages organisation

### CODE REVIEW

File: docs/global/Blocktime.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+
+### Block time
+
+When the protocol receives a new chain head, the block time (`blockperiodseconds`) and round
```
Comment (jframe) [inline docs/global/Blocktime.md:4]: For Qbft the round timer starts after the block time has expired not at the same time as the block timer.

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+
+    The `blockperiodseconds`, `blockreward`, and  `miningbeneficiary` properties
+    cannot be updated once your network is started.
+
+    We do not recommend changing `epochlength` in a running network. Changing the `epochlength`
+    after genesis can result in illegal blocks.
+
+The properties with specific values in the QBFT genesis files are:
+
+* `nonce` - `0x0`
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/HowTo/Configure/Consensus-Protocols/QBFT.md:107]: Done.

File: docs/Reference/CLI/CLI-Subcommands.md [hunk 3]
```diff
@@ -220,11 +222,15 @@ hexadecimal string.
     cat extra_data.json | besu rlp encode > rlp.txt
     ```
 
-The `IBFT_EXTRA_DATA` type is the only type supported for RLP encoding. The
-[IBFT 2.0 genesis file](../../HowTo/Configure/Consensus-Protocols/IBFT.md#genesis-file) includes
-the `IBFT_EXTRA_DATA` type.
+Supported types are:
 
-???+ summary "IBFT 2.0 Extra Data"
+* `IBFT_EXTRA_DATA` - The [IBFT 2.0 genesis file](../../HowTo/Configure/Consensus-Protocols/IBFT.md#genesis-file) includes
```
[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/Reference/CLI/CLI-Subcommands.md:227]: Done

File: docs/global/Blocktime.md (hunk 1 above)
Comment (bgravenorst) [reply docs/global/Blocktime.md:4]: I "think" my change makes sense, please let me know.

File: docs/Concepts/Consensus-Protocols/Comparing-PoA.md [hunk 4]
```diff
@@ -30,15 +30,15 @@ Properties to consider when comparing Clique and IBFT 2.0 are:
 
 ### Immediate finality
 
-IBFT 2.0 has immediate finality. When using IBFT 2.0 there are no forks and all valid blocks get
+IBFT 2.0 and QBFT has immediate finality; there are no forks and all valid blocks get
```
Comment (alexandratran) [inline docs/Concepts/Consensus-Protocols/Comparing-PoA.md:33]:
```suggestion
IBFT 2.0 and QBFT have immediate finality; there are no forks and all valid blocks get
```

File: docs/Concepts/Consensus-Protocols/Comparing-PoA.md [hunk 5]
```diff
@@ -50,9 +50,9 @@ redundancy if the validator fails.
 
 ### Liveness
 
-Clique is more fault tolerant than IBFT 2.0. Clique tolerates up to half of the validators failing.
-IBFT 2.0 networks require greater than or equal to two-thirds of validators to be operating to
-create blocks. For example, an IBFT 2.0 network of:
+Clique is more fault tolerant than IBFT 2.0 and QBFT. Clique tolerates up to half of the validators
+failing. IBFT 2.0 and QBFT networks require greater than or equal to two-thirds of validators to be
+operating to create blocks. For example, a IBFT 2.0 or QBFT network of:
```
Comment (alexandratran) [inline docs/Concepts/Consensus-Protocols/Comparing-PoA.md:55]:
```suggestion
operating to create blocks. For example, an IBFT 2.0 or QBFT network of:
```

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md [hunk 6]
```diff
@@ -0,0 +1,354 @@
...
+1. Reduce `requesttimeoutseconds` until you start to see round changes occurring.
+1. Increase `requesttimeoutseconds` to the value where round changes are no longer occurring.
+
+!!! tip
+
+    View [`TRACE` logs](../../../Reference/API-Methods.md#admin_changeloglevel) to see round change
+    log messages.
+{!global/Config-Options.md!}
+
+## Adding and removing validators
```
Comment (alexandratran) [inline docs/HowTo/Configure/Consensus-Protocols/QBFT.md:178]: I moved this section for Clique and IBFT 2.0 to the separate "Adding Validators" page in a recent PR (now merged). Should probably do the same for QBFT for consistency.

File: docs/Concepts/Consensus-Protocols/Comparing-PoA.md [hunk 7]
```diff
@@ -4,9 +4,9 @@ description: Besu  Proof of Authority consensus protocols comparison
...
 
-Besu implements the Clique and IBFT 2.0 Proof of Authority consensus protocols. Proof of Authority
-consensus protocols work when participants know each other and there is a level of trust between
-them. For example, in a permissioned consortium network.
+Besu implements the Clique, IBFT 2.0, and QBFT Proof of Authority consensus protocols. Proof of
+Authority consensus protocols work when participants know each other and there is a level of trust
+between them. For example, in a permissioned consortium network.
 
 Proof of Authority consensus protocols have faster block times and a much greater transaction
 throughput than the Ethash Proof of Work consensus protocol used on the Ethereum MainNet.
```
Comment (alexandratran) [inline docs/Concepts/Consensus-Protocols/Comparing-PoA.md:12]: The paragraph below this should mention QBFT.

**Review by alexandratran — COMMENTED** [review]
Looks good. Added a few comments.

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (jframe) [reply docs/HowTo/Configure/Consensus-Protocols/QBFT.md]: Sorry to be a bother but can the mixHash description be changed back to that fixed value again. I tested with Quorum and it turns out it has to be that fixed value and it is difficult to make the change in quorum so I have just raised a PR in Besu to change the behaviour back to use the fixed value again.

[the remark this answers is not in the document]
Comment (bgravenorst) [reply docs/HowTo/Configure/Consensus-Protocols/QBFT.md]: Done

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md (hunk 6 above)
Comment (bgravenorst) [reply docs/HowTo/Configure/Consensus-Protocols/QBFT.md:178]: Done

File: docs/Concepts/Consensus-Protocols/Comparing-PoA.md (hunk 7 above)
Comment (bgravenorst) [reply docs/Concepts/Consensus-Protocols/Comparing-PoA.md:12]: Done

File: docs/Concepts/Consensus-Protocols/Comparing-PoA.md (hunk 4 above)
Comment (bgravenorst) [reply docs/Concepts/Consensus-Protocols/Comparing-PoA.md:33]: Done.

File: docs/Concepts/Consensus-Protocols/Comparing-PoA.md (hunk 5 above)
Comment (bgravenorst) [reply docs/Concepts/Consensus-Protocols/Comparing-PoA.md:55]: Done

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
```
Comment (alexandratran) [inline docs/HowTo/Configure/Consensus-Protocols/QBFT.md:182]: Remove this header and the blank line below

File: docs/HowTo/Configure/Consensus-Protocols/Add-Validators.md [hunk 9]
```diff
@@ -252,3 +252,174 @@ To add or remove validators without voting:
...
     Specifying a transition block in the past could result in unexpected behavior.
+
+## QBFT
+
+Add and remove validators by [voting](#adding-and-removing-validators-by-voting) or if network
+conditions require it, [without voting](#adding-and-removing-validators-without-voting).
+
+### Adding and removing validators by voting
+
+To propose adding or removing validators using the JSON-RPC methods, enable the HTTP interface
```
Comment (alexandratran) [inline docs/HowTo/Configure/Consensus-Protocols/Add-Validators.md:263]: This paragraph is covered at the top of this page, so can be deleted here.

**Review by alexandratran — APPROVED** [review]
Other than two minor fixes, looks good to me.

File: docs/HowTo/Configure/Consensus-Protocols/Add-Validators.md [hunk 10]
```diff
@@ -252,3 +252,169 @@ To add or remove validators without voting:
...
+            ```
+
+1. Restart all nodes in the network using the updated genesis file.
+1. To verify the changes after the transition block, call
+   [`qbft_getValidatorsByBlockNumber`](../../../Reference/API-Methods.md#qbft_getvalidatorsbyblocknumber),
+   specifying `latest`.
+
+!!! caution
+    Do not specify a transition block in the past. Specifying a transition block in the past could
+    result in unexpected behaviour.
```
Comment (NicolasMassart) [inline docs/HowTo/Configure/Consensus-Protocols/Add-Validators.md:420]: Can you give any example of behaviour? Stopping the network, forking?

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+
+!!! example "One Initial Validator in `toEncode.json` file"
+
+    ```json
+    [
+     "9811ebc35d7b06b3fa8dc5809a1f9c52751e1deb"
+    ]
+    ```
+
+Copy the RLP encoded data to the `extraData` property in the genesis file.
```
Comment (NicolasMassart) [inline docs/HowTo/Configure/Consensus-Protocols/QBFT.md:140]: Can you add the RLP output for this example?

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+
+    View [`TRACE` logs](../../../Reference/API-Methods.md#admin_changeloglevel) to see round change
+    log messages.
+{!global/Config-Options.md!}
+
+<!-- Acronyms and Definitions -->
+
+*[Vanity]: Validators can include anything they like as vanity data.
+*[RLP]: Recursive Length Prefix.
+
```
Comment (NicolasMassart) [inline docs/HowTo/Configure/Consensus-Protocols/QBFT.md:186]: Can you add the full signification of QBFT acronym? I guess the Q is for Quorum.

File: mkdocs.yml [hunk 13]
```diff
@@ -221,7 +223,7 @@ nav:
 markdown_extensions:
     - toc:
         permalink: 
-        toc_depth: 3
+        toc_depth: 4
```
Comment (NicolasMassart) [inline mkdocs.yml:226]: This setting impacts all the doc site, I'm not sure it's good to change it. We should adapt the new content to fit in 3 levels.

**Review by NicolasMassart — COMMENTED** [review]
Did not check the tutorial yet

**Review by jframe — APPROVED** [review]

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md (hunk 12 above)
Comment (jframe) [reply docs/HowTo/Configure/Consensus-Protocols/QBFT.md:186]: QBFT doesn't stand for anything, it's just QBFT so we don't need to define it.

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md (hunk 11 above)
Comment (bgravenorst) [reply docs/HowTo/Configure/Consensus-Protocols/QBFT.md:140]: Done

**Pull request merged**